### ISSUE TITLE
Active Target is lost when closing window

### ISSUE THREAD

**Issue by TheStealthReporter** [noseglid/atom-build#427]
Every time I select an active target, close all Atom windows and open my project/session again, I have to manually reselect the old active target (it jumps back to default).

My expected behaviour is that this value is restored so I can pick up the work the way I left it.

I encountered this problem both with Make and Cargo (though it probably occurs with other build providers as well).

**Comment by oli-obk** [comment]
To allow this we'd need unique ids for targets, because the list of targets may change between reloads. Maybe we could hash the target's settings...

**Comment by noseglid** [comment]
Yes, this is "caused" by the core project, so the issue is correct here.

What could be possible to do is store the state when the package is deactivated (e.g. the *name* of the current active target) and restore that when the package is activated *after* the targets have been refreshed.

**Comment by jfs** [comment]
I would like to see this added as well.

As an alternative or stop-gap it would also be nice to be able to set a default target on a per-build-package basis. So for `build-gradle` I might select `build` as my default where someone else might select `classes`. For `build-rake` I might want to select a different default.

If the default target didn't exist you could fall back to the existing behavior.
